modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/InjectableParameterResolver.java

```diff
@@ -130,29 +128,54 @@ private BindingContext createBindingContext(ParameterValueResolver parameterValu
           DataType valueDataType = DataType.builder().type(value.getClass()).mediaType(mediaType).build();
           value = new TypedValue<>(value, valueDataType);
         }
-        bindingContextBuilder.addBinding(getWithBindingPrefix(parameterModel.getName()), (TypedValue) value);
+        componentParameters.put(parameterModel.getName(), (TypedValue) value);
       }
     }
+
+    bindingContextBuilder.addBinding(BINDING_IDENTIFIER,
+                                     new TypedValue(componentParameters, DataType.builder().mapType(Map.class).build()));
     return bindingContextBuilder.build();
   }
 
+  private Collection<String> componentParameterIdentifiers(BindingContext bindingContext) {
+    return ((Map) bindingContext.lookup(BINDING_IDENTIFIER).get().getValue()).keySet();
+  }
+
   private String getResolvedParameterValuesExpression(Collection<String> identifiers) {
     StringBuilder expression = new StringBuilder();
     expression.append(EXPRESSION_PREFIX);
     expression.append(
                       injectableParametersMap.values().stream()
                           .filter(injectableParameterInfo -> identifiers
-                              .contains(getWithBindingPrefix(getParameterNameFromExtractionExpression(injectableParameterInfo
-                                  .getExtractionExpression()))))
+                              .contains(getParameterNameFromExtractionExpression(injectableParameterInfo
+                                  .getExtractionExpression())))
                           .map(injectableParameterInfo -> "\""
                               + injectableParameterInfo.getParameterName() + "\"  : "
-                              + getWithBindingPrefix(injectableParameterInfo.getExtractionExpression()))
+                              + BINDING_IDENTIFIER_ACCESS
+                              + sanitizeExpression(injectableParameterInfo.getExtractionExpression()))
                           .collect(Collectors.joining(", ")));
     expression.append(EXPRESSION_SUFFIX);
 
     return expression.toString();
   }
 
+  private String sanitizeExpression(String extractionExpression) {
```

Comment:
I was wondering if this would work with "attributes" but I see it would:
```account.@type``` is how an attribute is accessed in DW.

Reply:
Good catch, I have just tried this. The extraction expression will have to be `account.@"type"`.

Reply:
I think that needs to be taken into account from the connector, since the SDK asks for a DW selector for the value and using type is not valid. I think the best effort made for the component parameter name is enough.

Reply:
Ok, makes sense. Nice work for documentation